server/src/test/java/org/opensearch/indices/recovery/RemoteStorePeerRecoverySourceHandlerTests.java

```diff
@@ -63,6 +66,15 @@ public void testReplicaShardRecoveryUptoLastFlushedCommit() throws Exception {
             // Step 7 - Check retention lease does not exist for the replica shard
             assertEquals(1, primary.getRetentionLeases().leases().size());
             assertFalse(primary.getRetentionLeases().contains(ReplicationTracker.getPeerRecoveryRetentionLeaseId(replica2.routingEntry())));
+
+            Directory storeDirectory = ((FilterDirectory) ((FilterDirectory) primary.store().directory()).getDelegate()).getDelegate();
+            ((BaseDirectoryWrapper) storeDirectory).setCheckIndexOnClose(false);
+
+            storeDirectory = ((FilterDirectory) ((FilterDirectory) replica1.store().directory()).getDelegate()).getDelegate();
+            ((BaseDirectoryWrapper) storeDirectory).setCheckIndexOnClose(false);
+
+            storeDirectory = ((FilterDirectory) ((FilterDirectory) replica2.store().directory()).getDelegate()).getDelegate();
+            ((BaseDirectoryWrapper) storeDirectory).setCheckIndexOnClose(false);
```

Comment:
Why are we preventing checks on directory close ? I see these checks are important as they validate store for readable commits before close.

Reply:
If we have unsynced files at time of closing index shard, those checks will force those files to be corrupted which will end up with all these tests failing with a corruption exception. 

Since, we are changing the behavior to remove commits, the files will not be part of a commit when remote store is enabled but this is expected behavior on our end so we set the checks to false. 

This check removal is a known way of dealing with this issue - also done in testRestoreShardFromRemoteStore previously.

Reply:
I think we should sync the files manually with `directory.sync()` before shard close here.  While we don't want to commit locally it would ensure those segments are durably persisted before close when opened/restarted would not need to fetch them again based on how remote sync is [implemented today](https://github.com/opensearch-project/OpenSearch/blob/e3542011cd9584c354405ba6cbf9a31ced7bc5ce/server/src/main/java/org/opensearch/index/shard/IndexShard.java#L4786) in indexShard - it does a cksum locally instead of using a replicationdiff.

Reply:
Note - we will want to extend this check also to SegmentReplicationDiff with https://github.com/opensearch-project/OpenSearch/issues/7802 to preserve this check against local file outside of the replication diff.

Reply:
@Poojita-Raj I was suggesting to sync the files in NRTReplicationEngine before close instead of the tests themselves then we can check index as normal + the segments would be durable on restart.